.github/workflows/release-chart.yaml

```diff
@@ -26,67 +27,59 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Configure Git
-        run: |
-          git config user.name "$GITHUB_ACTOR"
-          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
-
-      - name: Get chart version
-        id: chart_version
-        uses: mikefarah/yq@f03c9dc599c37bfcaf533427211d05e51e6fee64 # v4.47.1
-        with:
-          cmd: yq eval '.version' './charts/metrics-server/Chart.yaml'
-
-      - name: Get chart app version
-        id: chart_app_version
-        uses: mikefarah/yq@f03c9dc599c37bfcaf533427211d05e51e6fee64 # v4.47.1
-        with:
-          cmd: yq eval '.appVersion' './charts/metrics-server/Chart.yaml'
-
-      - name: Check can release
-        id: check_can_release
+      - name: Setup
+        id: setup
         run: |
           set -euo pipefail
 
-          branch_name="${GITHUB_REF##*/}"
-          app_version_prefix="${branch_name##*-}"
+          chart_file="./charts/metrics-server/Chart.yaml"
 
-          app_version_regex="^${app_version_prefix//./\.}"
+          chart_version="$(yq eval '.version' "${chart_file}")"
+          app_version="$(yq eval '.appVersion' "${chart_file}")"
+          release=false
 
-          chart_version_match="$(echo "${{ steps.chart_version.outputs.result }}" | grep -Po "^\d+\.\d+\.\d+$" || true)"
-          app_version_match="$(echo "${{ steps.chart_app_version.outputs.result }}" | grep -Po "^${app_version_prefix//./\.}" || true)"
-
-          if [[ -z "${chart_version_match}" ]] || [[ -z "${app_version_match}" ]]
-          then
-            echo "continue=false" >> $GITHUB_OUTPUT
-          else
-            echo "continue=true" >> $GITHUB_OUTPUT
+          if [[ -z "$(gh release view --repo kubernetes-sigs/metrics-server "metrics-server-helm-chart-${chart_version}" --json id || true)" ]]; then
+            branch_name='${{ github.ref_name }}'
+            if [[ "${branch_name}" == "master" ]]; then
+              release=true
+            else
+              branch_version_prefix="$(grep -Eo '[0-9]+\.[0-9]+$' <<<"${branch_name}")"
```

Comment:
I'm a little confused on why this is called branch_version_prefix when it looks like it is the branch_version. The prefix to me would be release-